src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,31 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+
+type Props = {
+  currentTodo: Todo;
+  onCurrentTodo: (todo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ currentTodo, onCurrentTodo }) => {
+  const [user, setUser] = useState<User | null>(null);
```

Comment:
You initialized your state with props, which is not recommended. This can lead to bugs and make the component harder to understand. Here is why: https://stackoverflow.com/a/50403930, https://reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state